common/types.go

```diff
@@ -301,6 +301,32 @@ func SetField(obj map[string]interface{}, k string, v interface{}) bool {
 	return true
 }
 
+// DelField detete a value in a tree based on dot key
+func DelField(obj map[string]interface{}, k string) interface{} {
+	components := strings.Split(k, ".")
+	o, ok := obj[components[0]]
+	if !ok {
+		return obj
+	}
+	if len(components) == 1 {
+		delete(obj, k)
+		return obj
+	}
+
+	object, ok := o.(map[string]interface{})
+	if !ok {
+		return obj
+	}
+	r := DelField(object, strings.SplitN(k, ".", 2)[1])
+	if len(r.(map[string]interface{})) != 0 {
+		obj[components[0]] = r
```

Comment:
I'm OK to remove it. I found this a bit weird because you want to delete an entry and it will delete the map, but maybe for metadata it is better to remove it.